app/solutions/routes.py

```diff
@@ -37,13 +41,84 @@ def get_general_solutions():
     they click the general solutions menu button. General solutions are ordered based
     on relevance predicted from users personal values.
     """
+    session_id = request.args.get("session-id")
+    if session_id:
+        user_scores = [np.array(get_scores_vector(session_id))]
+    else:
+        user_scores = None
+
+    if user_scores:
+        user_liberal, user_conservative = predict_radical_political(user_scores)
+    else:
+        user_liberal, user_conservative = None, None
+
     try:
         recommended_general_solutions = (
-            SOLUTION_PROCESSOR.get_user_general_solution_nodes()
+            SOLUTION_PROCESSOR.get_user_general_solution_nodes(
+                user_liberal, user_conservative
```

Comment:
As a next step, to make the integration we can think of ways to deal with conflicting predictions (e.g., a case where user_liberal ==1  && user_conservative==1).